org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/impl/ChangeSerializer.java

```diff
@@ -182,15 +185,15 @@ protected void endRecordChanges(IAcceptor<IEmfResourceChange> changeAcceptor, Su
 			}
 		}
 		
-		monitor.beginTask("Creating text changes...", updaters.size());
+		subMonitor = SubMonitor.convert(subMonitor, "Creating text changes...", updaters.size());
 		for (ResourceUpdater updater : updaters) {
-			monitor.split(1);
+			subMonitor.split(1);
 			updater.applyChange(deltas, changeAcceptor);
 		}
 		for (ResourceUpdater updater : updaters) {
 			updater.unload();
 		}
-		monitor.done();
+		subMonitor.done();
```

Comment:
this is now obsolete. it is guaranteed that all ticks in this submon are consumed, and anyway it is not necessary to call done then.

Reply:
but I'm fine with leaving it to make this more explicit. But I'd prefer to declare a second variable for the subprogress only and call on the "main" subMonitor done here then, i.e. leave the line as is here but assign the result of convert in l.188 to another variable.

Reply:
> this is now obsolete. it is guaranteed that all ticks in this submon are consumed, and anyway it is not necessary to call done then.

I don't think so, as subclasses (e.g. https://github.com/eclipse/xtext-eclipse/blob/387fbf08c8f34c5f50cae05229be59a8ec004949/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/edit/ChangeSerializerTextEditComposer.java#L65) call this method via the original API `endRecordChanges(IAcceptor<IEmfResourceChange>)` with `this.monitor` potentially being set to anything.

As said in #1130, according to https://www.eclipse.org/articles/Article-Progress-Monitors/article.html, Section `4.0 Responsibilities of callers and callees` this is valid.

> But I'd prefer to declare a second variable for the subprogress only

In theory I agree. The implementation, however, returns just the provided object in that case so I skipped the additional variable.

Reply:
TLDR; : Leave as is.

> may call this method via the original API endRecordChanges(IAcceptor<IEmfResourceChange>) with this.monitor potentially being set to anything.

Right. That's the reason why you create a SubMonitor here, and you could call `done()` on that are just leave it out. You are _not_ operating on `this.monitor` here. `this.monitor` could be anything, correct, and that's why it is always a good choice to create a SubMonitor first and operate only on that in a particular method.

> In theory I agree. The implementation, however, returns just the provided object in that case so I skipped the additional variable.

Right. Since you know that it is already a SubMonitor and you know a SubMonitor converted returns the same instance, a new variable is not necessary. The assignment is also not necessary then.

Reply:
> The assignment is also not necessary then.

Indeed. :-)